.cursor/rules/components/basic_component.mdc

```diff
@@ -0,0 +1,77 @@
+---
+description: Rules and checklist for creating a basic Langflow Component
+globs: 
+alwaysApply: false
+---
+# Rule: How to Create a Basic Langflow Component
+
+## Purpose
+Guide for Creating a Langflow Component
+
+---
+
+### 1. Gather Requirements
+
+Ask the user for:
+- **Component Name:** What should the component be called?
+- **Description:** What does the component do?
+- **Inputs:** What are the required inputs? (e.g., text, dropdown, boolean, etc.)
+- **Outputs:** What should the component output? (e.g., a message, a value, etc.)
+- **Category:** Which component category should this component be stored under in `langflow/src/backend/base/langflow/components`
+
+### 2. Define the Component
+
+- Inherit from `Component`.
+- Set `display_name`, `description`, `icon`, and `name`.
```

Comment:
Name should never be set because it is mainly to support components that their filenames were used as their name. Now the name is the class name.
```suggestion
- Set `display_name`, `description`, `icon`.
```